code/ATMOSPHERICS/components/binary_devices/circulator.dm

```diff
@@ -80,13 +80,13 @@
 		return
 	if (last_pressure_delta > 0 && recent_moles_transferred > 0)
 		if (temperature_overlay)
-			add_overlay(temperature_overlay)
+			overlays += temperature_overlay
```

Comment:
Please make sure to not use the overlay += for overlays as add_overlay is better in every way